src/com/facebook/buck/rules/args/WorkerMacroArg.java

```diff
@@ -99,6 +103,18 @@ public String getStartupArgs() {
     return workerTool.getArgs();
   }
 
+  public Optional<String> getPersistentWorkerKey() {
+    if (workerTool.isPersistent()) {
+      return Optional.of(buildTarget.toString());
```

Comment:
The key should probably include the target's cell in some form to avoid false sharing of workers between rules in different cells.